src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs

```diff
@@ -4697,7 +4700,7 @@ private static bool WaitAllCore(Task[] tasks, int millisecondsTimeout, Cancellat
                 {
                     // Add any exceptions for this task to the collection, and if it's wait
                     // notification bit is set, store it to operate on at the end.
-                    foreach (Task task in waitedOnTaskList)
+                    foreach (Task task in CollectionsMarshal.AsSpan(waitedOnTaskList))
```

Comment:
This is one of those things it'd be nice if the compiler could do for us.  @jaredpar, what would your tolerance be for a PR that changed foreach loops over `List<T>` to use `CollectionsMarshal.AsSpan` if it existed?  Could be extended in the future to do so for other types that have an AsSpan method as well.

Reply:
Isn't necessarily straightforward as it breaks the version check for enumerators; which for example ForEach externalizes (currently and with this change) in case `action` modifies the List:
https://github.com/dotnet/coreclr/blob/29906e719ca1b91cb9b762f916b9bfe93b2dc992/src/System.Private.CoreLib/shared/System/Collections/Generic/List.cs#L571-L578

It was one of the main reasons for it being in CollectionsMarshal rather than an extension

Reply:
Good point. I retract my question :)